### PR TITLE
Admin ticket

### PR DESCRIPTION
**Pull request by RiyaSantore04**

### CODE REVIEW

File: app/models/ticket.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class Ticket < ApplicationRecord
-  belongs_to :employee
+  belongs_to :employee, dependent: :destroy
```
Comment (salvimohan) [inline app/models/ticket.rb:2]: Don;t add dependent destroywith belongs_tp

File: app/views/admin_main/tickets/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    <tbody>
+        <%# if current_employee.role.name == 'Admin' %>
+      <% @tickets.each do |ticket| %>
+      <tr>
+        <th scope="row"><button type="button" class="btn btn-info btn-sm" data-toggle="modal" data-target="#myModal-<%=ticket.id%>"><%= raw ticket.description.slice(0,15)+ "..." %></button></th>
+        <td><%= ticket.employee.name %></td>
+        <td><%= ticket.ticket_type %></td>
+        <td><%= link_to "Show", admin_main_ticket_path(ticket) %></td>
+         <% if current_employee.role.name == 'Admin' %>
+        <td><% if ticket.status == 'accept' %>
```
Comment (salvimohan) [inline app/views/admin_main/tickets/index.html.erb:24]: move logic in helper

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Comment < ApplicationRecord
+  belongs_to :ticket
```
Comment (salvimohan) [inline app/models/comment.rb:2]: who comments so add association of employee as well

File: app/controllers/admin_main/tickets_controller.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	      @tickets = Ticket.where(employee_id: current_employee)
+	    end
+    end
+  
+    def show
+    	@ticket = Ticket.find_by_id params[:id]
+    end
+
+    def update
+	  	@ticket = Ticket.find(params[:id])
```
Comment (ror-uday) [inline app/controllers/admin_main/tickets_controller.rb:16]: correct indentation

File: app/models/ability.rb [hunk 5]
```diff
@@ -6,7 +6,7 @@ class Ability
   def initialize(employee)
     # Define abilities for the passed in user here. For example:
     #   user ||= User.new # guest user (not logged in)
-        if employee.role.name == "HR"
+        if employee.role.name == "HR" ||  employee.role.name == "Admin"
```
Comment (ror-uday) [inline app/models/ability.rb:9]: why are you not using instance method?

File: app/views/admin_main/tickets/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                  <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
+                  </div>
+                </div>
+
+              </div>
+            </div>
+          </div>
+          <% end %>
+        </tbody>
+
```
Comment (ror-uday) [inline app/views/admin_main/tickets/index.html.erb:67]: remove uncessary space and correct the alignment of code

**Review by ror-uday — CHANGES_REQUESTED** [review]

File: app/controllers/admin_main/tickets_controller.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
+class AdminMain::TicketsController < ApplicationController
+
+    def index 
+	    if current_user.role.name == 'Admin'
```
Comment (ror-uday) [inline app/controllers/admin_main/tickets_controller.rb:4]: why are you not using instance method?

File: app/views/admin_main/tickets/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+  <main id="main" class="main">
+		<%#= form_with(model: @ticket) do |form| %>
+		<%= form_with model: @ticket, :url => admin_main_ticket_path(ticket),  :method => "patch"  do |form| %>
+		  <div class="field">
+		    <%= form.label :add_reason%>
+		    <%= form.text_field :add_reason, class: "form-control mt-3 mb-3" %>
+		  </div>
+
+
```
Comment (ror-uday) [inline app/views/admin_main/tickets/edit.html.erb:11]: correct indentation

**Review by ror-uday — CHANGES_REQUESTED** [review]

File: app/controllers/admin_main/comments_controller.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+class AdminMain::CommentsController < ApplicationController
+
+	def create
+	    @ticket = Ticket.find_by_id params[:ticket_id]
+	    @comment = Comment.new( commenter: params[:commenter], ticket_id: params[:ticket_id])
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply app/controllers/admin_main/comments_controller.rb]: Changes Done

File: app/models/ticket.rb (hunk 1 above)
Comment (RiyaSantore04) [reply app/models/ticket.rb]: Changes Done

File: app/models/comment.rb (hunk 3 above)
Comment (RiyaSantore04) [reply app/models/comment.rb:2]: Changes Done

File: app/controllers/admin_main/tickets_controller.rb [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+	      @tickets = Ticket.order(created_at: :desc)
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply app/controllers/admin_main/tickets_controller.rb]: Changes Done

File: app/controllers/admin_main/tickets_controller.rb (hunk 4 above)
Comment (RiyaSantore04) [reply app/controllers/admin_main/tickets_controller.rb]: Changes Done

File: app/models/ability.rb (hunk 5 above)
Comment (RiyaSantore04) [reply app/models/ability.rb:9]: Changes Done

File: app/views/admin_main/tickets/index.html.erb (hunk 6 above)
Comment (RiyaSantore04) [reply app/views/admin_main/tickets/index.html.erb]: Changes Done

File: app/views/admin_main/tickets/edit.html.erb (hunk 8 above)
Comment (RiyaSantore04) [reply app/views/admin_main/tickets/edit.html.erb]: Changes Done

**Review by rorpyll — CHANGES_REQUESTED** [review]

File: app/controllers/admin_main/tickets_controller.rb [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+	    else
+	      @tickets = Ticket.where(employee_id: current_employee)
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply app/controllers/admin_main/tickets_controller.rb]: Changes Done

File: app/views/admin_main/tickets/index.html.erb [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+            <%= link_to 'Accepted', admin_main_ticket_path(ticket), method: :patch, class: 'btn btn-success disabled' %>
+          <% elsif ticket.status == 'decline' %>
+            <%= link_to 'Declined', admin_main_decline_tickett_path(ticket), method: :patch, class: 'btn btn-danger disabled' %>
+          <% else %>
+            <%= link_to 'Accept', admin_main_ticket_path(ticket), method: :patch, class: 'btn btn-success' %>
+            <%= link_to 'Decline', admin_main_decline_tickett_path(ticket), method: :patch, class: 'btn btn-danger' %>
+          <% end %>
+        </td>
+        <% else %>
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply app/views/admin_main/tickets/index.html.erb]: Changes Done

File: config/routes.rb [hunk 13]
```diff
@@ -18,8 +18,8 @@
   root 'dashboard#index'
   resources :performances
   resources :holidays
-  devise_for :admin_users, ActiveAdmin::Devise.config
-  ActiveAdmin.routes(self)
+  # devise_for :admin_users, ActiveAdmin::Devise.config
+  # ActiveAdmin.routes(self)
```
Comment (chiraggargsass) [inline config/routes.rb:22]: remove comments

**Review by chiraggargsass — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 13 above)
Comment (RiyaSantore04) [reply config/routes.rb:22]: Changes done

File: app/views/shared/_header.html.erb [hunk 14]
```diff
@@ -1,151 +1,150 @@
...
-          <% end %>
-      </li>
-    </ul>
-  </aside>
\ No newline at end of file
+    <li class="nav-item">
+      <% if current_employee.is_admin? %>
+        <%= link_to profile_path , class: "nav-link collapsed" do %>
+          <i class="bi bi-person"></i>
+          <%= 'Profile' %>
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply app/views/shared/_header.html.erb:69]: Changes done

File: app/views/tickets/index.html.erb [hunk 15]
```diff
@@ -14,30 +14,24 @@
                   </tr>
                 </thead>
                 <tbody>
+
+
```
[the remark this answers is not in the document]
Comment (RiyaSantore04) [reply app/views/tickets/index.html.erb:18]: changes done

**Pull request closed without merging**